subsys/net/ip/tcp.c

```diff
@@ -2476,28 +2476,80 @@ static enum net_verdict tcp_in(struct tcp *conn, struct net_pkt *pkt)
 		do_close = true;
 		break;
 	case TCP_FIN_WAIT_1:
-		/* Acknowledge but drop any data */
-		conn_ack(conn, + len);
+		/* Acknowledge but drop any data, but subtract any duplicate data */
+		if (th && (len > 0)) {
```

Comment:
This and all code below depends on `th` being set, I'd check it before and break early instead of duplicating `th` check in each if case.

Reply:
Good suggestion